library/core/src/main/java/com/google/android/exoplayer2/mediacodec/MediaCodecRenderer.java

```diff
@@ -1170,5 +1170,19 @@ private static boolean codecNeedsMonoChannelCountWorkaround(String name, Format
     return Util.SDK_INT <= 18 && format.channelCount == 1
         && "OMX.MTK.AUDIO.DECODER.MP3".equals(name);
   }
-
+  /**
+   * Returns whether the decoder is known to be non Adaptive.
+   * <p>
+   * If false is returned then we explicitly override codecIsAdaptive,
+   * setting it to false.
+   *
+   * @param name The decoder name.
+   * @param format The input format.
+   * @return False if the device is known  to be non adaptive .
+   */
+  private static boolean codecSupportsAdaptive(String name, Format format) {
+    return !(
+        (Util.SDK_INT == 19 && Util.MODEL.equals("ODROID-XU3")
```

Comment:
If this device ever had anything prior to API level 19, consider <= instead of ==. It was probably broken on earlier API levels too (unless you know otherwise).